docs/designs/evaluation_design.md

```diff
@@ -0,0 +1,63 @@
+
+Questions to discuss:
+* None of training checkpoints will be evaluated in above two solutions which are different from tf.estimator. tf.estimator will load the last checkpoint saved by train to evaluate in a separated pod named evaluator. 
+* Should master preserve the model to checkpoint after evaluation?
+
+## Evaluate by Loading Checkpoint to Pserver
+Generally, deep learning frameworks preserve model as checkpoint every many seconds or steps. So pserver can load the model from checkpoint for evaluation. There are also two designs for pserver to load model. 
+
+1. The master launches additional pods as evaluation pserver.\
+When an evaluation job starts, the master will launch additional pods as evaluation pservers and evaluation pservers will load the model from checkpoint.\
+The worker gets a model from evaluation pservers to inference outputs for evaluation tasks and reports outputs to master.\
+The master will release those pods after receiving all outputs of evaluation tasks from the worker.\
+In this case, evaluation is separated from train. So, some workers can process evaluation tasks and other workers can process train tasks at the same time. But more additional pods are needed.
+
+2. Existing pservers stop train task and load model from checkpoint to evaluate.\
+When an evaluation job starts, pservers preserve the current model to the temporary file and load model from the last checkpoint. After evaluation, pservers will restore the model from temporary file preserved and continue to train. \
+The design of pserver will be complicated in this case. What's more, some workers may be reporting gradients when the master change model to evaluation for psersers and pservers can not update model by those gradients at this moment.
+
+
+## Introduction to tf.estimator to Evaluate
+tf.estimator will launch pods as pservers, workers and evaluator when a distributed train job is submitted with ParameterServerStrategy. The evaluator does not participate in training process. tf.estimator decides when to evaluate by [throttle_secs](https://www.tensorflow.org/api_docs/python/tf/estimator/EvalSpec#throttle_secs). tf.estimator will check whether there are new checkpoints in checkpoint directory every throttle_secs after the last evaluation ends. The evaluator only has one instance and it will restore the whole model from the last checkpoint. So, the evaluation of tf.estimator is not distributed and not fault-tolerant.
+
+## Summary:
+1. If the difference with gradients of several mini-batches of model between workers to evaluate is acceptable, the first solution in `Evaluate with Current Model in Pserver` can be adopted. Otherwise，the master should send stop signals to pservers to stop update model after an evaluation job starts.
```

Comment:
If the difference with gradients of several mini-batches of model between workers to evaluate is acceptable => I think what you mean is "the inconsistency between model variables and embedding", not difference between workers.